nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-security/src/test/java/org/apache/nifi/web/security/request/ContentLengthFilterTest.java

```diff
@@ -0,0 +1,183 @@
+        Assert.assertTrue(StringUtils.containsIgnoreCase(response, "500 Server Error"));
+        Assert.assertTrue(StringUtils.containsIgnoreCase(response, "Timeout"));
+    }
+
+
+    @Test
+    public void testJettyMaxFormSize() throws Exception {
+        // This shows that the jetty server option for 'maxFormContentSize' is insufficient for our needs because it
+        // catches requests like this:
+        configureAndStartServer(readFormAttempt, FORM_CONTENT_SIZE);
+        String form = "a=" + StringUtils.repeat("1", FORM_CONTENT_SIZE);
+        String response = localConnector.getResponse(String.format(FORM_REQUEST, form.length(), form));
+        Assert.assertTrue(StringUtils.containsIgnoreCase(response, "413 Payload Too Large"));
+
+
+        // But it does not catch requests like this:
+        response = localConnector.getResponse(String.format(POST_REQUEST, form.length(), form+form));
+        Assert.assertTrue(StringUtils.containsIgnoreCase(response, "417 Read Too Many Bytes"));
+    }
+
+
+
+    private HttpServlet readFullyAndRespondOK = new HttpServlet() {
+        @Override
+        protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+            ServletInputStream input = req.getInputStream();
+            while (!input.isFinished()) {
+                input.read();
+            }
+            resp.setStatus(HttpServletResponse.SC_OK);
+        }
+    };
+
+
+    private HttpServlet readFormAttempt = new HttpServlet() {
+        @Override
+        protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+            try {
+                req.getParameterMap();
+                ServletInputStream input = req.getInputStream();
+                int count = 0;
+                while (!input.isFinished()) {
+                    input.read();
+                    count += 1;
+                }
+                if (count > FORM_CONTENT_SIZE + "a=\n".length()) {
+                    resp.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, "Should not reach this code.");
+                } else {
+                    resp.sendError(HttpServletResponse.SC_EXPECTATION_FAILED, "Read Too Many Bytes");
+                }
+
+            } catch (final Exception e) {
+                // This is the jetty context returning a 400 from the maxFormContentSize setting:
+                if (StringUtils.containsIgnoreCase(e.getCause().toString(), "Form Too Large")) {
+                    resp.sendError(HttpServletResponse.SC_REQUEST_ENTITY_TOO_LARGE, "Payload Too Large");
+                } else {
+                    resp.sendError(HttpServletResponse.SC_FORBIDDEN, "Should not reach this code, either.");
+                }
+            }
+        }
+    };
+
+
+    private void configureAndStartServer(HttpServlet servlet, int maxFormContentSize) throws Exception {
```

Comment:
This should be moved above the test methods.

Reply:
So moved.